logs_sdk/lib/opentelemetry/sdk/logs/logger_provider.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # The SDK implementation of OpenTelemetry::Logs::LoggerProvider.
+      class LoggerProvider < OpenTelemetry::Logs::LoggerProvider
+        attr_reader :resource, :log_record_processors
+
+        UNEXPECTED_ERROR_MESSAGE = 'unexpected error in ' \
+          'OpenTelemetry::SDK::Logs::LoggerProvider#%s'
+        # Returns a new LoggerProvider instance.
+        #
+        # @param [optional Resource] resource The resource to associate with
+        #   new LogRecords created by {Logger}s created by this LoggerProvider.
+        # @param [optional Array] log_record_processors The
+        #   {LogRecordProcessor}s to associate with this LoggerProvider.
+        #
+        # @return [OpenTelemetry::SDK::Logs::LoggerProvider]
+        def initialize(
+          resource: OpenTelemetry::SDK::Resources::Resource.create,
+          log_record_processors: []
+        )
+          @log_record_processors = log_record_processors
+          @mutex = Mutex.new
+          @resource = resource
+          @stopped = false
+        end
+
+        # Creates an {OpenTelemetry::SDK::Logs::Logger} instance.
+        #
+        # @param [optional String] name Instrumentation package name
+        # @param [optional String] version Instrumentation package version
+        #
+        # @return [OpenTelemetry::SDK::Logs::Logger]
+        def logger(name = nil, version = nil)
```

Comment:
This signature is inconsistent between `TracerProvider` and `MeterProvider`. In the latter, we used:
```ruby
def meter(name, version: nil)
```
so `name` is required and `version` is a keyword arg. IIRC we're going to have to add one or two more args at some point 🔜 , so keywords for optional args are vastly preferred.

We should go back and fix this for `TracerProvider`. I _think_ that can be done in a backwards-compatible fashion.

For `MeterProvider`, we opted to require `name`. If a user explicitly passes `nil` or `""`, we can handle it, but those are both considered invalid - I'd rather have an explicitly invalid required arg than an invalid default arg.

Reply:
Sounds great. I prefer the keyword args, I just defaulted to what was in TracerProvider since that was already stable. 

Updated in: ca7c15f5bbb2834e339fb723bfe24667fe2035a3